### PR TITLE
Allow disabling of ICS31 Cross Chain Queries

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #4040

## Description

This PR adds a configuration which disables the relaying of ICS31 Cross Chain Queries

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by romac** [comment]
Reminder to add the new option to `config.toml`

File: crates/relayer-cli/src/chain_registry.rs [hunk 1]
```diff
@@ -168,6 +168,7 @@ where
         compat_mode: None,
         clear_interval: None,
         excluded_sequences: BTreeMap::new(),
+        allow_ccq: true,
```
Comment (ancazamfir) [inline crates/relayer-cli/src/chain_registry.rs:171]: Could you also add with some comments in `config.toml`?

**Review by ancazamfir — APPROVED** [review]
Looks good, thanks @ljoss17 ! Maybe we should open an issue  or discussion to review, evaluate, consolidate the filtering features.

File: crates/relayer-cli/src/chain_registry.rs (hunk 1 above)
Comment (ljoss17) [reply crates/relayer-cli/src/chain_registry.rs:171]: Sure, if it looks good I'll finish writing the documentation and changelog entry

**Comment by ljoss17** [comment]
> Looks good, thanks @ljoss17 ! Maybe we should open an issue or discussion to review, evaluate, consolidate the filtering features.
Sure, we can discuss in this issue https://github.com/informalsystems/hermes/issues/4040

**Comment by ljoss17** [comment]
> Looks good, thanks @ljoss17 ! Maybe we should open an issue or discussion to review, evaluate, consolidate the filtering features.

Sure I'll open an issue for the filtering. I also found out these methods which could be either removed or used, https://github.com/informalsystems/hermes/blob/v1.9.0/crates/relayer/src/supervisor.rs#L317

File: crates/relayer-cli/src/chain_registry.rs (hunk 1 above)
Comment (ljoss17) [reply crates/relayer-cli/src/chain_registry.rs:171]: Done in 17081fec528a95b0ffc8e130c4af36ce70fea8c0

**Comment by ljoss17** [comment]
> Looks good, thanks @ljoss17 ! Maybe we should open an issue or discussion to review, evaluate, consolidate the filtering features.

I opened this issue https://github.com/informalsystems/hermes/issues/4052

File: crates/relayer/src/worker.rs [hunk 2]
```diff
@@ -209,16 +210,24 @@ pub fn spawn_worker_tasks<ChainA: ChainHandle, ChainB: ChainHandle>(
...
-                cross_chain_query.clone(),
-            );
-            task_handles.push(cross_chain_query_task);
-
-            (Some(cmd_tx), None)
+            if config.chains.iter().any(|chain| match chain {
+                ChainConfig::CosmosSdk(c) => {
+                    (c.id == cross_chain_query.dst_chain_id) && c.allow_ccq
+                }
+            }) {
```
Comment (romac) [inline crates/relayer/src/worker.rs:217]: Can we add an `allow_ccq` method on `ChainConfig` to simplify the code a bit?

```suggestion
            if config.chains.iter().any(|chain| chain.id() == cross_chain_query.dst_chain_id) && chain.allow_ccq()) {
```

**Review by romac — CHANGES_REQUESTED** [review]
Awesome, just left one nitpick

File: crates/relayer/src/worker.rs (hunk 2 above)
Comment (ljoss17) [reply crates/relayer/src/worker.rs]: Done in 9d67e17f486652bcfb07609c74ed635fdb966473

**Review by romac — APPROVED** [review]

**Pull request merged**